types/Color.php

```diff
@@ -56,15 +56,16 @@ public function renderMultiValue($values, \Doku_Renderer $R, $mode) {
     /**
      * @inheritDoc
      */
-    public function valueEditor($name, $rawvalue) {
+    public function valueEditor($name, $rawvalue, $htmlID) {
         $class = 'struct_color';
         if(!preg_match('/^#[a-f0-9]{6}$/', $rawvalue)) {
             // any non-color (eg. from a previous type) should default to the default
             $rawvalue = $this->config['default'];
         }
         $name = hsc($name);
         $rawvalue = hsc($rawvalue);
-        $html = "<input name=\"$name\" value=\"$rawvalue\" class=\"$class\" type=\"color\" />";
+        $id = !empty($htmlID) ? "id=\"$htmlID\"" : '';
+        $html = "<input name=\"$name\" value=\"$rawvalue\" class=\"$class\" type=\"color\" $id />";
```

Comment:
Those lines (in all of the files) are starting to get ugly. I wonder if it would be better to switch to a `buildAttributes()` approach where where we use associative arrays to prepare the HTML attributes. It would also do the empty($htmlID) check for us.